src/main/java/user_use_cases/UserDatabaseGateway.java

```diff
@@ -0,0 +1,76 @@
+package user_use_cases;
+
+import entities.User;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+
+public class UserDatabaseGateway {
+    // User Data is saved in order: first column -> username, second column -> password
+
+    private static final String NAME_OF_USER_DATABASE = "src/main/java/Databases/UserDatabase.csv";
+
+
+    public Boolean userExists(String username) {
+        try {
+            File file = new File(NAME_OF_USER_DATABASE);
+            Scanner scanner = new Scanner(file);
+            String delimiter = ",";
+            String line;
+
+            while(scanner.hasNext()){
+                line = scanner.nextLine();
```

Comment:
Should include this while loop in the constructor and save all the users in a hash map or list so that you don't have to run this loop every time, you can just search the list